### ISSUE TITLE
vagrant box not starting for v8

### ISSUE THREAD

**Issue by mirjan-hoffmann** [TIBHannover/edu-sharing-box#40]
Hi @Edmondi-Kacaj,

With one of your last changes, you implemented to stop the old tomcat before you install edu-sharing v8. It seems that this also tries to do so, if there is no old edu-sharing-version:

```
TASK [edu-sharing : register oersi - add app-oersi.properties.xml] *************
skipping: [edu-sharing-vm]

TASK [edu-sharing : register oersi - edit applicationfiles in ccapp-registry.properties.xml] ***
skipping: [edu-sharing-vm]

TASK [edu-sharing : Ensure tomcat is stopped.] *********************************
fatal: [edu-sharing-vm]: FAILED! => {"changed": true, "cmd": "/home/vagrant/bin/tomcat.sh stop", "delta": "0:00:00.002538", "end": "2023-05-02 12:27:53.809592", "msg": "non-zero return code", "rc": 127, "start": "2023-05-02 12:27:53.807054", "stderr": "/bin/sh: 1: /home/vagrant/bin/tomcat.sh: not found", "stderr_lines": ["/bin/sh: 1: /home/vagrant/bin/tomcat.sh: not found"], "stdout": "", "stdout_lines": []}
```

Could you adjust this please and don't stop an old edu-sharing version, if there is no such version?

**Comment by mirjan-hoffmann** [comment]
another error occurs later in the playbook:

```
TASK [edu-sharing : Clear optional properties in client config.] ***************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ImportError: No module named lxml
failed: [edu-sharing-vm] (item=/config/values/register/recoverUrl) => {"ansible_loop_var": "item", "changed": false, "item": "/config/values/register/recoverUrl", "msg": "Failed to import the required Python library (lxml) on edu-sharing's Python /usr/bin/python. Please read module documentation and install in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter"}
```

Seems like the corresponding package (`python-lxml`) isn't installed for edu-sharing-new

**Comment by mirjan-hoffmann** [comment]
Unfortunately, a clean `vagrant up` still don't works:

```
TASK [Get the path to the edu-sharing cluster docker volume.] ******************
fatal: [edu-sharing-vm]: FAILED! => {"changed": true, "cmd": "sg docker -c \"docker volume inspect --format '{{.Mountpoint}}'  $(docker volume ls -q |grep '_repository-service-volume-config-cluster')\"\n", "delta": "0:00:00.054597", "end": "2023-05-03 09:47:30.361069", "msg": "non-zero return code", "rc": 1, "start": "2023-05-03 09:47:30.306472", "stderr": "permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get \"http://%2Fvar%2Frun%2Fdocker.sock/v1.24/volumes\": dial unix /var/run/docker.sock: connect: permission denied\n\"docker volume inspect\" requires at least 1 argument.\nSee 'docker volume inspect --help'.\n\nUsage:  docker volume inspect [OPTIONS] VOLUME [VOLUME...]\n\nDisplay detailed information on one or more volumes", "stderr_lines": ["permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get \"http://%2Fvar%2Frun%2Fdocker.sock/v1.24/volumes\": dial unix /var/run/docker.sock: connect: permission denied", "\"docker volume inspect\" requires at least 1 argument.", "See 'docker volume inspect --help'.", "", "Usage:  docker volume inspect [OPTIONS] VOLUME [VOLUME...]", "", "Display detailed information on one or more volumes"], "stdout": "", "stdout_lines": []}
...
TASK [edu-sharing : Clear optional properties in client config.] ***************
failed: [edu-sharing-vm] (item=/config/values/register/recoverUrl) => {"ansible_loop_var": "item", "changed": false, "item": "/config/values/register/recoverUrl", "msg": "The target XML source '/var/lib/docker/volumes/edu_sharing_docker_version_repository-service-volume-config-node/_data/defaults/client.config.xml' does not exist."}
failed: [edu-sharing-vm] (item=/config/values/banner) => {"ansible_loop_var": "item", "changed": false, "item": "/config/values/banner", "msg": "The target XML source '/var/lib/docker/volumes/edu_sharing_docker_version_repository-service-volume-config-node/_data/defaults/client.config.xml' does not exist."}
failed: [edu-sharing-vm] (item=/config/values/banner) => {"ansible_loop_var": "item", "changed": false, "item": "/config/values/banner", "msg": "The target XML source '/var/lib/docker/volumes/edu_sharing_docker_version_repository-service-volume-config-node/_data/defaults/client.config.xml' does not exist."}
failed: [edu-sharing-vm] (item=/config/values/allowedLicenses) => {"ansible_loop_var": "item", "changed": false, "item": "/config/values/allowedLicenses", "msg": "The target XML source '/var/lib/docker/volumes/edu_sharing_docker_version_repository-service-volume-config-node/_data/defaults/client.config.xml' does not exist."}
failed: [edu-sharing-vm] (item=/config/values/loginUrl) => {"ansible_loop_var": "item", "changed": false, "item": "/config/values/loginUrl", "msg": "The target XML source '/var/lib/docker/volumes/edu_sharing_docker_version_repository-service-volume-config-node/_data/defaults/client.config.xml' does not exist."}
ok: [edu-sharing-vm] => (item=/config/values/loginAllowLocal)
ok: [edu-sharing-vm] => (item=/config/values/loginProvidersUrl)
ok: [edu-sharing-vm] => (item=/config/values/loginProviderTargetUrl)
ok: [edu-sharing-vm] => (item=/config/values/logout)
ok: [edu-sharing-vm] => (item=/config/values/workspaceViewType)
```

@Edmondi-Kacaj  this seems to be the problem (which we discussed already some weeks ago) that the docker-group is not available in `$(...)` for the first time the script is running. We already found a solution for this (see https://github.com/TIBHannover/edu-sharing-box/blob/master/ansible/roles/edu-sharing/tasks/edu-sharing-docker/client-configuration.yml#L4)
Could you fix https://github.com/TIBHannover/edu-sharing-box/blob/master/ansible/roles/edu-sharing/tasks/edu-sharing-docker/edu-sharing-config-file.yml#L4 too please?

**Comment by Edmondi-Kacaj** [comment]
Hello @mirjan-hoffmann 

Yes looks like this is already fixed in (https://github.com/TIBHannover/edu-sharing-box/blob/master/ansible/roles/edu-sharing/tasks/edu-sharing-docker/client-configuration.yml#L4), but I found another similar problem and I created a pull request to fix that too, see #44

**Comment by mirjan-hoffmann** [comment]
now a new box is starting again
